packages/contracts/contracts/discovery/pools/Pool.sol

```diff
@@ -50,32 +81,54 @@ abstract contract Pool is IPool, RisingTide {
         override(IPool)
         onlyProject
     {
-        if (investorBalances[_investor] == 0) {
+        if (investors[_investor].uncappedAllocation == 0) {
             investorByIndex[_investorCount] = _investor;
             _investorCount++;
         }
 
-        investorBalances[_investor] += _amount;
+        investors[_investor].uncappedAllocation += _amount;
         totalUncappedAllocations += _amount;
+        IERC20(investmentToken).safeTransferFrom(
+            _investor,
+            address(this),
+            _amount
+        );
     }
 
     function setIndividualCap(uint256 _cap) external {
         _risingTide_setCap(_cap);
     }
 
     /// @inheritdoc IPool
-    function refund(address _to) external override(IPool) {
-        revert("not yet implemented");
+    function refund(address _to) external override(IPool) capCalculated {
+        Investor storage investor = investors[_to];
+        require(!investor.refunded, "already refunded");
+
+        uint256 amount = refundAmount(_to);
+        require(amount > 0, "No tokens to refund");
+
+        investor.refunded = true;
+        IERC20(investmentToken).transfer(_to, amount);
+
+        emit Refund(_to, amount);
     }
 
     /// @inheritdoc IPool
     function refundAmount(address _to)
-        external
+        public
         view
         override(IPool)
-        returns (uint256 amount)
+        returns (uint256)
     {
-        revert("not yet implemented");
+        // TODO: Should ignore rising tide if project lost
+        if (!risingTide_isValidCap()) {
+            return 0;
+        }
+
+        uint256 uncapped = investors[_to].uncappedAllocation;
+        uint256 capped = allocation(_to);
+
+        return uncapped - capped;
```

Comment:
isn't this mixing currencies? I'm assuming `allocation` and `uncappedAllocation` is specified in project tokens?

Reply:
Ah, true. Need to use the rate. Probably should implement function to go both ways in the Project, no?